### PR TITLE
Resolve import collisions

### PR DESCRIPTION
**Pull request by ahmadj-levelbenefits**
It is common to have a file that imports two files with exports that have symbol collisions. Here is a minimal reproduction of that use case: see import-collision.ts for a demonstration.

I'm not sure how to fix this: ts-proto uses ts-poet for import generation, but the dependency hasn't been updated in a while: v1.0.1 is completely different from ts-poet at v3.2.

How should this be resolved?

The only real option I see is updating, to 3.2: https://github.com/stephenh/ts-poet/blob/master/src/Code.ts#L123

### CODE REVIEW

File: integration/import-collision/import-collision.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
+import { Job } from './aws/aws';
+import { Job } from './gcp/gcp';
```
Comment (ahmadj-levelbenefits) [inline integration/import-collision/import-collision.ts:2]: Here's the issue.

**Comment by stephenh** [comment]
Hi @ahmadj-levelbenefits , yep, your diagnosis is exactly correct: this is a known-issue in ts-poet 1.x, it's solved "for free" in ts-poet 3.x, and the solution for ts-proto is to just bite the bullet and covert ts-proto over to use ts-poet 3.x.

In general ts-poet 3.x is a much lighter-weight / "less Java" API, so the migration shouldn't be hard/complex, just tedious. :-)

I've been generally procrastinating this for "someday". Would be awesome if you wanted to take a crack at it, and I'd be happy to help with any pointers/etc.

**Comment by ahmadj-levelbenefits** [comment]
To get the migration to be gradual, would it be possible to alias the old version, and use the new version from now on? https://classic.yarnpkg.com/en/docs/cli/add/#toc-yarn-add-alias

This way we can integration test at many steps of the migration instead of one big one at the end, at the risk of the migration taking net longer, but this specific problem being fixed much quicker (this is the major blocker towards us using it internally).

This might not be possible to begin with: since 3.2 expects to work with its notion of nodes for interleaving.

**Comment by stephenh** [comment]
@ahmadj-levelbenefits "since 3.2 expects to work with its notion of nodes for interleaving." right, the two ts-poets can't really interop, so the only way to move gradually would be `file-a.ts`-is-generated-by-v1 and `file-b.ts`-is-generated-by-v2. However, ts-proto's open is generally a single admittedly huge file, so I think that effectively keeps an all-at-once migration.

**Comment by ahmadj-levelbenefits** [comment]
I see this issue: https://github.com/stephenh/ts-proto/issues/36 effectively tracks this suggestion. 

I unfortunately can't commit to this effort, so I'll close this PR. Thanks for your words, work, and time.

**Pull request closed without merging**